src/modeler/AutoRest.Swagger/Validation/AvoidMsdnReferences.cs

```diff
@@ -21,6 +21,18 @@ public class AvoidMsdnReferences : TypedRule<string>
         /// </summary>
         public override ValidationCategory ValidationCategory => ValidationCategory.SDKViolation;
 
+        /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
```

Comment:
So the way I look at it is, 
- Default - All rules that could be applied to a simple OpenAPI document 
- ARM - All rules speced out by the ARM folks
- Data-Plane - All rules applicable to the data-plane specs

Something like this is applicable to both ARM and Data-plane but isn't necessarily a Default, in which case we could apply multiple flags to the validation rule.